File: webapp/src/models/Player.ts
export class Player {
    uuid: string;
    name: string;
    violationLevel: number;
    banDate?: Date;


    constructor(uuid: string, name: string, violationLevel: number, banDate?: Date) {
        this.uuid = uuid;
        this.name = name;
        this.violationLevel = violationLevel;
        this.banDate = banDate;
    }

    public static fromJson(json: any): Player {
        return new Player(json.uuid, json.name, json.violationLevel, json.banDate);
    }
}

File: src/main/java/com/cimeyclust/ncpwebserver/models/Player.java
package com.cimeyclust.ncpwebserver.models;

import java.sql.Date;
import java.sql.Timestamp;

public class Player {
    public String uuid;
    public String name;
    public Timestamp banDate;
    public double violationLevel;

    public Player(String uuid, String name, Timestamp  banDate, double violationLevel) {
        this.uuid = uuid;
        this.name = name;
        this.banDate = banDate;
        this.violationLevel = violationLevel;
    }
}
